File: rotateArr/rotateArr.py
def rotate(nums, k):
    n = len(nums)
    if n == 0 or k == 0:
        return nums

    k = k % n
    if k < 0:
        k += n

    nums[:] = nums[::-1]
    nums[:k] = nums[:k][::-1]
    nums[k:] = nums[k:][::-1]
    return nums

# Example usage
nums = [1, 2, 3, 4, 5, 6, 7]
k = 3
rotate(nums, k)
print(nums)

File: rotateArr/rotateArr.cpp
#include <iostream>
#include <vector>
#include <algorithm> // For std::reverse

void rotate(std::vector<int>& nums, int k) {
    int n = nums.size();
    if (n == 0 || k == 0) return;

    k = k % n;
    if (k < 0) k += n;

    std::reverse(nums.begin(), nums.end());
    std::reverse(nums.begin(), nums.begin() + k);
    std::reverse(nums.begin() + k, nums.end());
}

int main() {
    std::vector<int> nums = {1, 2, 3, 4, 5, 6, 7};
    int k = 3;
    rotate(nums, k);
    for (int num : nums) {
        std::cout << num << " ";
    }
    return 0;
}
